### PR TITLE
Fat 5575 create imageforproducts

### PR DESCRIPTION
**Pull request by Activict**

### CODE REVIEW

File: StoreApp/Controllers/StoreController.cs [hunk 1]
```diff
@@ -106,7 +107,47 @@ public ActionResult CreateProduct(CreateProductViewModel product)
...
             }
 
-            productService.Create(productDTO);
+            if (file != null && file.ContentLength > 0)
+            {
+                string pictureType = file.ContentType.ToLower();
+
+                if (pictureType != "image/jpg" &&
+                    pictureType != "image/jpeg" &&
+                    pictureType != "image/png")
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/StoreController.cs]: I propose to add these types to the config file. For example, we can add the list of the extensions to <appSettings> section of webconfig file

File: .gitignore [hunk 2]
```diff
@@ -337,4 +337,5 @@ ASALocalRun/
 .localhistory/
 
 # BeatPulse healthcheck temp database
-healthchecksdb
\ No newline at end of file
+healthchecksdb
+/StoreApp/Pictures
```
Comment (AleksandroBoko) [inline .gitignore]: As I could understand you added the Picture folder to the file .gitignore. According to it I cannot have an access to this folder cause it wasn't downloaded to the remote branch. And of course, I am not able to see the pictures.

File: StoreApp/Controllers/StoreController.cs [hunk 3]
```diff
@@ -106,7 +107,47 @@ public ActionResult CreateProduct(CreateProductViewModel product)
...
+                int productId = productService.GetAll()
+                                              .FirstOrDefault(p => p.Name.Equals(productDTO.Name) &&
+                                                                   p.BrandId.Equals(productDTO.BrandId) &&
+                                                                   p.ProducerId.Equals(productDTO.ProducerId)).Id;
+
+                var pathStringPictures = new DirectoryInfo(string.Format($"{Server.MapPath(@"\")}Pictures\\Products"));
+                var pathStringProductsById = Path.Combine(pathStringPictures.ToString(), productId.ToString());
+
+                if (!Directory.Exists(pathStringProductsById))
+                    Directory.CreateDirectory(pathStringProductsById);
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/StoreController.cs]: It's better to use the style like this 
**if(condition)
{
    operations...
}**

File: StoreApp/Controllers/StoreController.cs [hunk 4]
```diff
@@ -106,7 +107,47 @@ public ActionResult CreateProduct(CreateProductViewModel product)
...
+
+                var pathSavePicture = string.Format($"{pathStringProductsById}\\{file.FileName}");
+                var pathSavePicturePreview = string.Format($"{pathStringProductsById}\\{ "preview_" + file.FileName}");
+
+                WebImage pic = new WebImage(file.InputStream);
+                pic.Resize(200, 200).Crop(1, 1);
+                pic.Save(pathSavePicture);
+                pic.Resize(24, 24).Crop(1, 1);
+                pic.Save(pathSavePicturePreview);
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/StoreController.cs]: I suppose we can optimize this block of code
For example we can create a new method SaveImage(param1, param2, param3) which handles the process of saving pictures. So the logic of saving will be separated from the controller. Think about it. If you want we can discuss this point.

File: StoreApp/Controllers/StoreController.cs [hunk 5]
```diff
@@ -156,7 +197,52 @@ public ActionResult EditProduct(EditProductViewModel product)
...
             }
 
-            productService.Edit(productDTO);
+            if (file != null && file.ContentLength > 0)
+            {
+                string pictureType = file.ContentType.ToLower();
+
+                if (pictureType != "image/jpg" &&
+                    pictureType != "image/jpeg" &&
+                    pictureType != "image/png")
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/StoreController.cs]: The same as before

File: StoreApp/Controllers/StoreController.cs [hunk 6]
```diff
@@ -156,7 +197,52 @@ public ActionResult EditProduct(EditProductViewModel product)
...
+                }
+
+                var pathSavePicture = string.Format($"{pathStringProductsById}\\{file.FileName}");
+                var pathSavePicturePreview = string.Format($"{pathStringProductsById}\\{ "preview_" + file.FileName}");
+
+                WebImage pic = new WebImage(file.InputStream);
+                pic.Resize(200, 200).Crop(1, 1);
+                pic.Save(pathSavePicture);
+                pic.Resize(24, 24).Crop(1, 1);
+                pic.Save(pathSavePicturePreview);
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/StoreController.cs]: Also, we can optimize it. Just create a new file service to handle this process

File: StoreApp/Controllers/StoreController.cs (hunk 1 above)
Comment (Activict) [reply StoreApp/Controllers/StoreController.cs]: fixed

File: StoreApp/Controllers/StoreController.cs (hunk 5 above)
Comment (Activict) [reply StoreApp/Controllers/StoreController.cs]: fixed

File: .gitignore (hunk 2 above)
Comment (Activict) [reply .gitignore]: fixed

File: StoreApp/Controllers/StoreController.cs (hunk 3 above)
Comment (Activict) [reply StoreApp/Controllers/StoreController.cs]: fixed

File: StoreApp/Controllers/StoreController.cs (hunk 4 above)
Comment (Activict) [reply StoreApp/Controllers/StoreController.cs]: fixed

File: StoreApp/Controllers/StoreController.cs (hunk 6 above)
Comment (Activict) [reply StoreApp/Controllers/StoreController.cs]: fixed

**Pull request merged**